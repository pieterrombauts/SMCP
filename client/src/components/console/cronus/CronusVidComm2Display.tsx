import React from 'react';
import styled from 'styled-components';

interface CronusVC2DProps {
  className?: string;
};

const UCronusVidComm2Display: React.FC<CronusVC2DProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_5.png'></image>
        <rect x="650" y="630" rx="5" ry="5" width="370" height="160" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>
        
        <text x="51%" y="68%" fill="white" fontSize="17" fontFamily="Verdana">1)</text>
        <text x="54%" y="68%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">P1 Upper Outboard (P1UPOB)</text>

        <text x="51%" y="71%" fill="white" fontSize="17" fontFamily="Verdana">2)</text>
        <text x="54%" y="71%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">P1 Lower Outboard (P1LOOB)</text>

        <text x="51%" y="74%" fill="white" fontSize="17" fontFamily="Verdana">3)</text>
        <text x="54%" y="74%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">Labs Starboard (LABS)</text>

        <text x="51%" y="77%" fill="white" fontSize="17" fontFamily="Verdana">4)</text>
        <text x="54%" y="77%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">0E</text>  
      </svg>
    </div>
  )
};

const CronusVidComm2Display = styled(UCronusVidComm2Display)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusVidComm2Display;