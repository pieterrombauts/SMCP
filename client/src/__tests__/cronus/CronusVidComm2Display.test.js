import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import CronusVidComm2Display from '../../components/console/cronus/CronusVidComm2Display'
import { shallow, mount, render} from 'enzyme';

describe("Cronus UHF communications display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<CronusVidComm2Display />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})