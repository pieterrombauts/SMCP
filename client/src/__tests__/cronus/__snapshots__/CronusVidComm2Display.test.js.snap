// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cronus UHF communications display renders correctly 1`] = `
.c0 svg {
  width: 100%;
  height: 100%;
}

<Styled(UCronusVidComm2Display)>
  <UCronusVidComm2Display
    className="c0"
  >
    <div
      className="c0"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 1300 990"
      >
        <image
          height={990}
          href="/media/Cronus_5.png"
          width={1300}
          x={0}
          y={0}
        />
        <rect
          height="160"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#36485b",
              "stroke": "#2a3a3f",
              "strokeWidth": "5",
            }
          }
          width="370"
          x="650"
          y="630"
        />
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="17"
          x="51%"
          y="68%"
        >
          1)
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="54%"
          y="68%"
        >
          P1 Upper Outboard (P1UPOB)
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="17"
          x="51%"
          y="71%"
        >
          2)
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="54%"
          y="71%"
        >
          P1 Lower Outboard (P1LOOB)
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="17"
          x="51%"
          y="74%"
        >
          3)
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="54%"
          y="74%"
        >
          Labs Starboard (LABS)
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="17"
          x="51%"
          y="77%"
        >
          4)
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="54%"
          y="77%"
        >
          0E
        </text>
      </svg>
    </div>
  </UCronusVidComm2Display>
</Styled(UCronusVidComm2Display)>
`;
